benches/fastx.rs

```diff
@@ -0,0 +1,191 @@
+#![feature(test)]
+
+extern crate test;
+
+use rand::distributions::Alphanumeric;
+use rand::{thread_rng, Rng};
+use std::io;
+use std::io::prelude::*;
+use tempfile::NamedTempFile;
+use test::{black_box, Bencher};
+
+use bio::io::{fasta, fastx};
+
+const BASES: &[u8] = b"ACTG";
+const ID_LEN: usize = 10;
+const DESC_LEN: usize = 20;
+const SEQ_LEN: usize = 100;
+const FASTA_SIZE: usize = 100;
+
+const ITERS: usize = 4000;
+
+fn gen_random_fasta() -> io::Result<io::Cursor<Vec<u8>>> {
+    let mut file = NamedTempFile::new()?;
+    let mut w = fasta::Writer::to_file(file.path())?;
+    let mut rng = thread_rng();
+    for _ in 0..FASTA_SIZE {
+        let id: String = thread_rng()
+            .sample_iter(&Alphanumeric)
+            .take(ID_LEN)
+            .map(char::from)
+            .collect();
+
+        let desc: String = thread_rng()
+            .sample_iter(&Alphanumeric)
+            .take(DESC_LEN)
+            .map(char::from)
+            .collect();
```

Comment:
Wait you create a RNG generator but you didn't use it it to generate `id` and `description` why ?

I also recommend usage of a seedable RNG in benchmark to have more stable result between benchmark.

To get this you need change Cargo.toml:
```
rand = { version = "0.8", features = ["std_rng"] }
```
And change line 25 in:
```
let mut rng = rand::rngs::StdRng::seed_from_u64(42);
```